lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -858,6 +860,77 @@ previewers.highlights = defaulter(function(_)
   }
 end, {})
 
+previewers.pickers = defaulter(function(_)
+  local ns_telescope_multiselection = vim.api.nvim_create_namespace "telescope_mulitselection"
+  local get_row = function(picker, preview_height, index)
+    if picker.sorting_strategy == "ascending" then
+      return index - 1
+    else
+      return preview_height - index
+    end
+  end
+  return previewers.new_buffer_previewer {
+
+    dyn_title = function(_, entry)
+      return entry.value.preview_title
+    end,
+
+    get_buffer_by_name = function(_, entry)
+      return tostring(entry.value.prompt_bufnr)
+    end,
+
+    teardown = function(self)
+      if self.state and self.state.last_set_bufnr and vim.api.nvim_buf_is_valid(self.state.last_set_bufnr) then
+        vim.api.nvim_buf_clear_namespace(self.state.last_set_bufnr, ns_telescope_multiselection, 0, -1)
+      end
+    end,
+
+    define_preview = function(self, entry, status)
+      putils.with_preview_window(status, nil, function()
+        local ns_telescope_entry = vim.api.nvim_create_namespace "telescope_entry"
+        local preview_height = vim.api.nvim_win_get_height(status.preview_win)
+        if not self.state.bufname then
```

Comment:
Can we flip this to make it a little bit more readable. `if self.state.bufname then return end`. I currently having a hard time with the indents and all the ends :laughing: